ksp_plugin/interface.cpp

```diff
@@ -372,16 +368,13 @@ void principia__DeserializePlugin(char const* const serialization,
   }
 
   // Decode the hexadecimal representation.
-  std::uint8_t const* const hexadecimal =
-      reinterpret_cast<std::uint8_t const*>(serialization);
-  int const hexadecimal_size = serialization_size;
-  auto bytes = HexadecimalDecode({hexadecimal, hexadecimal_size});
+  auto bytes = HexadecimalDecode({serialization, serialization_size});
   auto const bytes_size = bytes.size;
   (*deserializer)->Push(std::move(bytes));
 
   // If the data was empty, delete the deserializer.  This ensures that
   // |*plugin| is filled.
-  if (bytes_size == 0) {
+  if (bytes.size == 0) {
```

Comment:
This is UB.  Exercise for the reader: find why.  Hint: there was a reason for the intermediate variable.

Reply:
Done.